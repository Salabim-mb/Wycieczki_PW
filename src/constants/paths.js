export const paths = {
	DASHBOARD: {
		name: "Strona główna",
		path: "/",
	},
	BLOG: {
		name: 'Blog',
		path: '/:blog',
	},
	BLOG_CATEGORIES: {
		name: "Kategorie",
		path: "/blog/:category",
		redirect: (newCategory) => `/blog/${newCategory}`,
	},
	BLOG_ENTRY: {
		name: "Wpis",
		path: "/blog/:category/:entry_id",
		redirect: (category, entryId) => `/blog/${category}/${entryId}`,
	},
	POST_CREATOR: {
		name: 'Kreator wpisów',
		path: '/blog/post-creator/:id?',
		redirect: (entryId) => `/blog/post-creator/${entryId}`
	},
	CREATOR_CATEGORY: {
		name: "Krator kategorii",
		path: "/blog/category-creator/:id?",
		redirect: (categoryId) => `/blog/category-creator/${categoryId}`
	}
}