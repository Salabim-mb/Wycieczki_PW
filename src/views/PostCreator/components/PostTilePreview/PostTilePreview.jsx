import React from 'react';
import PropTypes from 'prop-types';
import { Accordion, AccordionSummary, AccordionDetails, Typography } from '@material-ui/core';
import { ExpandMore } from '@material-ui/icons';
import { PostTile } from 'components';

/* eslint-disable no-useless-escape */

const PostTilePreview = ({ cover, title, summary }) => (
	<Accordion>
		<AccordionSummary
			expandIcon={<ExpandMore />}
			aria-controls="panel1a-content"
			id="panel1a-header"
		>
			<Typography>PodglÄ…d kafelka posta</Typography>
		</AccordionSummary>
		<AccordionDetails>
			<PostTile
				cover={typeof cover === 'string' ? cover : URL.createObjectURL(cover)}
				title={title}
				summary={summary.replace(/(<\/?[^>]+(>|$))|(\&nbsp;)/g, '')}
			/>
		</AccordionDetails>
	</Accordion>
);

PostTilePreview.propTypes = {
	cover: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({ name: PropTypes.string })]),
	title: PropTypes.string,
	summary: PropTypes.string,
};

PostTilePreview.defaultProps = {
	cover: null,
	title: '',
	summary: '',
};

export default PostTilePreview;
