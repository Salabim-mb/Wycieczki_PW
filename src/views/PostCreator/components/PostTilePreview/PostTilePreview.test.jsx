import React from 'react';
import { render } from '@testing-library/react';
import paths from 'constants/api';
import PostTilePreview from './PostTilePreview';

jest.mock('components', () => ({
	// eslint-disable-next-line react/prop-types
	PostTile: ({ cover, title, summary }) => (
		<div>
			<img src={cover} alt="img" />
			<p>{title}</p>
			<p>{summary}</p>
		</div>
	),
}));

global.URL.createObjectURL = jest.fn().mockImplementation(() => '/link');

describe('PostTilePreview', () => {
	let props;

	beforeEach(() => {
		props = {
			cover: '/link',
			title: 'tytuÅ‚ kafla',
			summary: 'opis',
		};
	});

	it('should match snapshot', () => {
		const { container } = render(<PostTilePreview {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should match snapshot with file', () => {
		props.cover = new File(['foo'], 'foo.jpg', {
			type: 'image/jpg',
		});

		const { container } = render(<PostTilePreview {...props} />);

		expect(container).toMatchSnapshot();
	});
});
