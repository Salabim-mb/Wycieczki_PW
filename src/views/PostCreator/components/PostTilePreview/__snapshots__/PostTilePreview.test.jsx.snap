// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTilePreview should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Podgląd kafelka posta
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <div>
                <img
                  alt="img"
                  src="/link"
                />
                <p>
                  tytuł kafla
                </p>
                <p>
                  opis
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostTilePreview should match snapshot with file 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Podgląd kafelka posta
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <div>
                <img
                  alt="img"
                />
                <p>
                  tytuł kafla
                </p>
                <p>
                  opis
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
